src/colmap/geometry/gps.h

```diff
@@ -93,14 +93,27 @@ struct PosePrior {
 
   Eigen::Vector3d position =
       Eigen::Vector3d::Constant(std::numeric_limits<double>::quiet_NaN());
+  Eigen::Matrix3d position_covariance =
+      Eigen::Matrix3d::Constant(std::numeric_limits<double>::quiet_NaN());
   CoordinateSystem coordinate_system = CoordinateSystem::UNDEFINED;
 
   PosePrior() = default;
   explicit PosePrior(const Eigen::Vector3d& position) : position(position) {}
   PosePrior(const Eigen::Vector3d& position, const CoordinateSystem system)
       : position(position), coordinate_system(system) {}
-
-  inline bool IsValid() const { return position.array().isFinite().any(); }
+  PosePrior(const Eigen::Vector3d& position, const Eigen::Matrix3d& covariance)
+      : position(position), position_covariance(covariance) {}
+  PosePrior(const Eigen::Vector3d& position,
+            const Eigen::Matrix3d& covariance,
+            const CoordinateSystem system)
+      : position(position),
+        position_covariance(covariance),
+        coordinate_system(system) {}
+
+  inline bool IsValid() const { return position.allFinite(); }
```

Comment:
Last minor comment: we need to add bindings for the new constructors, attribute, and methods in pycolmap/geometry/bindings.cc - thank you!

Reply:
:+1: